quickwit-doc-mapping/src/default_doc_mapper/field_mapping_entry.rs

```diff
@@ -0,0 +1,605 @@
+                entries
+                    .iter()
+                    .map(|entry| {
+                        if let Some(child) = object.get(&entry.name) {
+                            entry.parse(child)
+                        } else {
+                            // TODO: define if we need to raise an error or not.
+                            Ok(vec![])
+                        }
+                    })
+                    .collect::<Result<Vec<_>, _>>()?
+                    .into_iter()
+                    .flatten()
+                    .map(|(path, entry)| (path.with_parent(&self.name), entry))
+                    .collect()
+            }
+            _ => {
+                return Err(DocParsingError::ValueError(
+                    self.name.clone(),
+                    "object type only support json object".to_owned(),
+                ))
+            }
+        };
+        Ok(parsed_values)
+    }
+
+    fn check_if_multi_values_allowed(&self, options: &IntOptions) -> Result<(), DocParsingError> {
+        let cardinality = options
+            .get_fastfield_cardinality()
+            .unwrap_or(Cardinality::SingleValue);
+        if cardinality != Cardinality::MultiValues {
+            return Err(DocParsingError::MultiValuesNotSupported(self.name.clone()));
+        }
+        Ok(())
+    }
+}
+
+/// A field path composed by the list
+/// of path components.
+/// Used to build a tantivy valid field name
+/// by joining its components with a
+/// special string `__dot__` as currently
+/// tantivy does not support `.`.
+pub struct FieldPath<'a> {
+    components: Vec<&'a str>,
+}
+
+impl<'a> FieldPath<'a> {
+    pub fn new(path: &'a str) -> Self {
+        Self {
+            components: vec![path],
+        }
+    }
+
+    /// Returns the FieldPath with the additionnal parent component.
+    /// This will consume your `FieldPath`.
+    pub fn with_parent(mut self, parent: &'a str) -> Self {
+        if !parent.is_empty() {
+            self.components.insert(0, parent);
+        }
+        self
+    }
+
+    // Returns a tantivy compatible field name.
+    pub fn tantivy_field_name(&self) -> String {
+        // Some components can contains dots.
+        self.field_name().replace(".", TANTIVY_DOT_SYMBOL)
+    }
+
+    // Returns field name built by joining its components with a `.` separator.
+    pub fn field_name(&self) -> String {
+        // Some components can contains dots.
+        self.components.join(".")
+    }
+}
+
+// Struct used for serialization and deserialization
+// Main advantage: having a flat structure and gain flexibility
+// if we want to add some syntaxic sugar in the mapping.
+// Main drawback: we have a bunch of mixed parameters in it but
+// seems to be reasonable.
+#[derive(Clone, Serialize, Deserialize)]
+struct FieldMappingEntryForSerialization {
+    name: String,
+    #[serde(rename = "type")]
+    type_with_cardinality: String,
+    #[serde(default = "default_as_true")]
+    stored: bool,
```

Comment:
ah ok so you defaulted the stored attribute to true at the field level...
ok for this.

Reply:
I think there is a misunderstanding about the stored stuff. we need to discuss that tomorrow.